toolkit/scripts/srpm_pack.mk

```diff
@@ -38,23 +38,27 @@ $(BUILD_SRPMS_DIR): $(STATUS_FLAGS_DIR)/build_srpms.flag
 	@echo Finished updating $@
 
 ifeq ($(DOWNLOAD_SRPMS),y)
+
+.SILENT: $(STATUS_FLAGS_DIR)/build_srpms.flag
+
 $(STATUS_FLAGS_DIR)/build_srpms.flag: $(local_specs) $(local_spec_dirs) $(SPECS_DIR)
 	for spec in $(local_specs); do \
 		spec_file=$${spec} && \
-		srpm_file=$$(rpmspec -q $${spec_file} --srpm --define='with_check 1' --define='dist $(DIST_TAG)' --queryformat %{NAME}-%{VERSION}-%{RELEASE}.src.rpm) && \
+		srpm_file=$$(rpmspec -q $${spec_file} --srpm --define='with_check 1' --define='dist $(DIST_TAG)' --queryformat %{NAME}-%{VERSION}-%{RELEASE}.src.rpm 2>/dev/null) && \
 		for url in $(SRPM_URL_LIST); do \
 			wget $${url}/$${srpm_file} \
 				-O $(BUILD_SRPMS_DIR)/$${srpm_file} \
-				--no-verbose \
+				-q \
 				$(if $(TLS_CERT),--certificate=$(TLS_CERT)) \
 				$(if $(TLS_KEY),--private-key=$(TLS_KEY)) \
 				&& \
 			touch $(BUILD_SRPMS_DIR)/$${srpm_file} && \
 			break; \
-		done || $(call print_error,Loop in $@ failed) ; \
-		{ [ -f $(BUILD_SRPMS_DIR)/$${srpm_file} ] || \
-			$(call print_error,Failed to download $${srpm_file});  } \
-	done || $(call print_error,Loop in $@ failed) ; \
+		done && echo "Downloaded $${url}/$${srpm_file}"; \
+	done || $(call print_error,Failed to download $${srpm_file}) ; \
+	echo "Removing empty (failed) SRPMS: "
```

Comment:
>echo "Removing empty (failed) SRPMS: " [](http://example.com/codeflow?start=1&length=38)

Are empty SRPMs expected? If so, why call them "failed"? If not, we shouldn't be printing this message if there were no failed SRPMs to avoid showing confusing log messages to the users (they might think that there's something wrong even if everything is fine). Also, will the build stop in case of such failures? I'm not sure what scenarios this is covering.

Reply:
After offline discussion, I will add a flag to trigger the behavior on or off.